zipline/algorithm.py

```diff
@@ -608,6 +608,11 @@ def _calculate_universe(self):
         # our universe is all the assets passed into `run`.
         return self._assets_from_source
 
+    def _compute_pipelines(self):
```

Comment:
Couple notes on this method:

1. Since we intend to override this to be called in a subclass downstream, we should make this a public method and document it. 
2. I would probably name this something like `compute_eager_pipelines` to make it clearer that this only computes pipelines that were attached with `eager=True`.
3. It doesn't look like anything is actually calling this in Zipline right now. I would expect that we want to call this at the start of `TradingAlgorithm.before_trading_start`. Note that that function exits early if `self._before_trading_start` is None, but we should still run eager pipelines even if the user hasn't defined a BTS, since part of the purpose of this PR is to remove the need for superfluous BTS implementations that just call pipeline_output and copy results onto context. 
4. We need a test for the new behavior here. Testing this is a bit tricky, since the values exposed to the user shouldn't change; only when they get computed. One way to test that things are computed in the correct order would be to create an algorithm that logs in its `before_trading_start` and during the execution of a `CustomFactor`. You can then capture the logs using a logbook TestHandler and verify that the logs from the CustomFactor fire before the logs from before_trading_start. (You could also do something similar by appending to a list on context instead of logging).